### DISCUSSION TITLE
Do not ask for a password at login

### DISCUSSION THREAD

**Discussion by sogeil** [ltsp/ltsp#843] [Q&A]
Hello!

A username should be able to be logged in without a password.

If I go to
> menu
-> Settings
--> Users and Groups
---> USERXY
----> Change password
-----> enable "Do not ask for a password at login"

then the password field disappears in the client's login window and instead there is a "Login" button, but when I click on it, a black screen appears and after about 2 seconds I'm back on the login screen.

How do you do it right?

Greeting,
Olaf

bodhi 6, LXDE & Moksha

**Comment by alkisg** [comment]
LTSP logins happen over SSH and allowing empty passwords for SSH isn't a good idea.

It's best for the users to HAVE a password, and to specify it in ltsp.conf, so that they're able to login without knowing or typing the password.

Read the man page about how to specify the passwords:
https://ltsp.org/man/ltsp.conf/#PASSWORDS_x

**Comment by sogeil** [comment]
Hello alkisg!

Sorry, maybe I didn't describe it well, or the google translator didn't understand me :-)

Only a single username XYZ should work without a password. This username will be logged in 99% of the time, the "real" users usually only briefly use software that runs in the browser and has its own login.

Some users sometimes need access to Office data (NAS/Samba). For this, everyone has their own "full" account + password on the server and the NAS with which they

a) can connect to the NAS in the session of user XYZ in Thunar

or

b) User XYZ logs out -> logs in with his own account + password -> NAS is then automatically integrated via fstab -> works -> then logs out again -> logs in user XYZ again (which is then used to work in the browser again).

It would be even nicer if user XYZ was automatically logged in after e.g. 15 seconds without another login when the login screen was called up, but I would be happy if his login would work without a password for the time being.
Maybe a guest account would be the solution, if I can configure it.

Greetings,
Olaf

**Comment by alkisg** [comment]
Hi Olaf,

I believe I understood you correctly. You need the following ltsp.conf:

```
[clients]
AUTOLOGIN="^pc\([0-9]*\)$/guest\1"
RELOGIN_TIMEOUT=15
PASSWORDS_GUESTS="guest[0-9-]*/cXdlcjEyMzQK"
```

So if you name your PCs like pc01, pc02, pc03, ...
and your guest users guest01, guest02, guest03, ...
and you set the guest password to `qwer1234`, which is `cXdlcjEyMzQK` in base64,
then autologin and relogin will work after 15 seconds.

During these 15 seconds, non-guest users may login normally.

I know you asked for a "single XYZ user", but that is NOT the correct way to do it. Applications aren't designed to work when multiple sessions happen using the same /home/guest directory. So the correct way is to use one guest account for each PC.

(There are also alternative ways to do it correctly, but I already wrote a lot :D)

**Comment by sogeil** [comment]
Hello alkisg!

Thank you for the explanation.
Since I'm not sure if I understand everything correctly with your "automatic" (?) code lines, I did a simple test with an existing account "Wohnb1" first, password changed to qwer1234:

[xx:xx:xx:xx:xx:xx]
HOSTNAME=PC-Wb1
AUTOLOGIN=Wohnb1
RELOGIN_TIMEOUT=15
PASSWORDS_PC-Wb1="Wohnb1/cXdlcjEyMzQK"

Unfortunately that doesn't work. The hostname is assumed by the client, but nothing else happens.
I installed LXDE, Moksha, icewm and openbox if that is relevant. Ultimately, LXDE should be used.

By the way, the page  https://gc.de/gc/base64/  shows me a different base64 value for qwer1234 (cXdlcjEyMzQ=). However, it doesn't work with that either.

Greetings,
Olaf

**Comment by alkisg** [comment]
Dashes `-` aren't allowed in ltsp.conf variable names, so your ltsp.conf is invalid as it contains `PASSWORDS_PC-Wb1`.

Change that to `PASSWORDS_PC_WB1`. If it still won't work, then when the client is in the login screen, use epoptes or ssh to access it, and send the output of `ps faux` and the contents of whatever of the following files exist:

```
/etc/gdm3/daemon.conf
/etc/lightdm/lightdm.conf
/etc/sddm.conf
/etc/lightdm/slick-greeter.conf
```

Regarding the encoded password, you're right, I forgot to omit the newline. The second one is the correct one, so use that:

```
$ echo qwer1234 | base64
cXdlcjEyMzQK
$ echo -n qwer1234 | base64
cXdlcjEyMzQ=
```

**Comment by sogeil** [comment]
Hello,

even after changing the name it doesn't work.
Unfortunately, I can't connect to the client via ssh either:

Client -> login -> terminal -> ifconfig -> client-IP -> logout

Server -> terminal -> ssh me@client-IP -> ssh: connect to host client-IP port 22: Connection refused

Perhaps these descriptions have something to do with the problem?:
https://bodhilinux.boards.net/thread/190/login-solved
https://www.linuxquestions.org/questions/linux-general-1/how-to-setup-passwordless-gui-login-on-bodhi-4175597466/
If so, that should presumably be in ltsp.conf, but I don't know how exactly.

**Comment by alkisg** [comment]
Install epoptes to be able to do what I said in my previous post. See https://ltsp.org/docs/installation/ for instructions.

**Comment by sogeil** [comment]
**/etc/lightdm/slick-greeter.conf**`
[Greeter]
background=/usr/share/backgrounds/bodhi/bodhi-login.png
theme-name=MokshaArcGreen
icon-theme-name=MokshaArcGreen
draw-user-backgrounds=true

**/etc/sddm.conf**
[Users]
MinimumUid=990

[Autologin]
User=bodhi
Session=enlightenment.desktop

**/etc/lightdm/lightdm.conf**
\# Generated by `ltsp init`, see man:ltsp(8)
\# You can append content here by specifying the LIGHTDM_CONF parameter
[Seat:*]
\# Work around https://github.com/CanonicalLtd/lightdm/issues/49
greeter-show-manual-login=true
greeter-hide-users=false
autologin-user=
autologin-user-timeout=15

**ps faux**
root         656  0.0  0.0      0     0 ?        I    20:06   0:00  \_ [kworker/
root         668  0.0  0.0      0     0 ?        I    20:06   0:00  \_ [kworker/
root         776  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [irq/127-
root         826  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [cryptd]
root         891  0.0  0.0      0     0 ?        I    20:06   0:00  \_ [kworker/
root         904  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [cifsiod]
root         928  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [smb3decr
root         931  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [cifsfile
root         932  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [cifsoplo
root         938  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [cifs-dfs
root         971  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         972  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         973  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         975  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         976  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         977  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         980  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root         982  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [cifsd]
root        1233  0.0  0.0      0     0 ?        I<   20:06   0:00  \_ [iprt-VBo
root        1240  0.0  0.0      0     0 ?        S    20:06   0:00  \_ [iprt-VBo
root        1383  0.0  0.0      0     0 ?        I    20:21   0:00  \_ [kworker/
root        1385  0.0  0.0      0     0 ?        I    20:21   0:00  \_ [kworker/
root        1396  0.0  0.0      0     0 ?        I    20:21   0:00  \_ [kworker/
root           1  0.1  0.1 103476 11916 ?        Ss   20:06   0:01 /sbin/init
root         641  0.0  0.1  36156  8396 ?        S<s  20:06   0:00 /lib/systemd/
root         652  0.0  0.0   5388   172 ?        Ss   20:06   0:00 /usr/sbin/blk
root         664  0.0  0.0  23920  7092 ?        Ss   20:06   0:00 /lib/systemd/
systemd+     672  0.0  0.0  18408  7724 ?        Ss   20:06   0:00 /lib/systemd/
systemd+     703  0.0  0.0  90260  6160 ?        Ssl  20:06   0:00 /lib/systemd/
systemd+     706  0.0  0.1  23892 12472 ?        Ss   20:06   0:00 /lib/systemd/
root         708  0.0  0.1 236160  8060 ?        Ssl  20:06   0:00 /usr/lib/acco
avahi        709  0.1  0.0   8616  3592 ?        Ss   20:06   0:00 avahi-daemon:
avahi        732  0.0  0.0   8352   328 ?        S    20:06   0:00  \_ avahi-dae
message+     710  0.0  0.0   7960  5068 ?        Ss   20:06   0:00 /usr/bin/dbus
root         711  0.0  0.2 259560 19044 ?        Ssl  20:06   0:00 /usr/sbin/Net
root         718  0.0  0.0  81960  3924 ?        Ssl  20:06   0:00 /usr/sbin/irq
root         720  0.0  0.0 421612   920 ?        Ssl  20:06   0:00 /usr/sbin/oss
root         721  0.0  0.0  16692  7624 ?        Ss   20:06   0:00 /lib/systemd/
root         722  0.0  0.1 123732 10312 ?        Ssl  20:06   0:00 /usr/sbin/the
root         724  0.0  0.1 392392 11912 ?        Ssl  20:06   0:00 /usr/lib/udis
root         725  0.0  0.0  13688  4960 ?        Ss   20:06   0:00 /sbin/wpa_sup
root         769  0.0  0.1 234836  9292 ?        Ssl  20:06   0:00 /usr/lib/poli
clamav       900  0.1  0.2 135264 17296 ?        Ss   20:06   0:01 /usr/bin/fres
root         901  0.0  0.0   7312  6012 ?        Ss   20:06   0:00 socat -T 60 o
root        1086  0.0  0.0   2616  1844 ?        S    20:06   0:00  \_ epoptes-c
root        1388  0.0  0.0   7052  3592 ?        S    20:21   0:00      \_ socat
root        1390  0.0  0.0   6896  3216 ?        S    20:21   0:00          \_ s
root        1391  0.0  0.0   7072  2844 ?        Ss   20:21   0:00             S
root        1392  0.0  0.0   7956  4916 pts/2    Ss   20:21   0:00             b
root        1397  0.0  0.0   9184  3644 pts/2    R+   20:22   0:00             p
root        1206  0.0  0.0   6860  3052 ?        Ss   20:06   0:00 /usr/sbin/cro
root        1214  0.0  0.0  19080   104 ?        Sl   20:06   0:00 /usr/lib/epso
daemon      1220  0.0  0.0   3800  2400 ?        Ss   20:06   0:00 /usr/sbin/atd
root        1230  0.0  0.0 305880  7452 ?        Ssl  20:06   0:00 /usr/sbin/lig
root        1239  0.2  0.9 908000 78836 tty7     Ssl+ 20:06   0:02  \_ /usr/lib/
root        1271  0.0  0.1 160048  8084 ?        Sl   20:06   0:00  \_ lightdm -
lightdm     1291  0.0  0.0   2616   540 ?        Ss   20:06   0:00  |   \_ /bin/
lightdm     1292  0.2  0.6 601904 53236 ?        Sl   20:06   0:02  |       \_ /
root        1348  0.0  0.0  11952  5840 ?        S    20:06   0:00  \_ lightdm -
root        1232  0.0  0.0   2972  1904 tty1     Ss+  20:06   0:00 /sbin/agetty 
lightdm     1276  0.0  0.1  18432  9772 ?        Ss   20:06   0:00 /lib/systemd/
lightdm     1277  0.0  0.0 104860  3224 ?        S    20:06   0:00  \_ (sd-pam)
lightdm     1282  0.0  0.2 280984 17864 ?        Ssl  20:06   0:00  \_ /usr/bin/
lightdm     1293  0.0  0.0   7248  4472 ?        Ss   20:06   0:00  \_ /usr/bin/
lightdm     1299  0.0  0.0 156128  5576 ?        Sl   20:06   0:00  \_ /usr/libe
lightdm     1321  0.0  0.0 237184  7708 ?        Ssl  20:06   0:00  \_ /usr/libe
lightdm     1326  0.0  0.0 312808  6488 ?        Sl   20:06   0:00  \_ /usr/libe
lightdm     1314  0.0  0.0 305368  6916 ?        Sl   20:06   0:00 /usr/libexec/
lightdm     1319  0.0  0.0   7116  4096 ?        S    20:06   0:00  \_ /usr/bin/
root        1333  0.0  0.1 249672  9720 ?        Ssl  20:06   0:00 /usr/lib/upow
lightdm     1352  0.0  0.0 162836  7556 ?        Sl   20:06   0:00 /usr/libexec/
root        1371  0.0  0.0   7072  2744 ?        Ss   20:07   0:00 SCREEN bash -
root        1372  0.0  0.0   7956  4932 pts/1    Ss   20:07   0:00  \_ bash -l
root        1377  0.0  0.0   9692  4824 pts/1    S+   20:07   0:00      \_ sudo 
root        1378  0.0  0.0  13788  7368 pts/1    S+   20:07   0:00          \_ j
root@PCWb1:~# 


I have also tested
LIGHTDM_CONF="greeter-show-manual-login=false"
LIGHTDM_CONF="greeter-hide-users=true"
LIGHTDM_CONF="autologin-user=Wohnbereich1"
LIGHTDM_CONF="autologin-user-timeout=5"

but only the last one is copied to the client's /etc/lightdm/lightdm.conf.

**Comment by alkisg** [comment]
The `autologin-user=` line is empty in your generated lightdm.conf.
This either means you didn't provide the `AUTOLOGIN=Wohnbereich1` line, or that the `Wohnbereich1` user doesn't exist in the booted client.

To provide many lines in an ltsp.conf variable, use the following syntax:

```
LIGHTDM_CONF="greeter-show-manual-login=false
greeter-hide-users=true
autologin-user=Wohnbereich1
autologin-user-timeout=5"
```

**Comment by sogeil** [comment]
Thank you, I had actually made a mistake, confused name and username.

**etc/ltsp/ltsp.conf**
[clients]
[xx:xx:xx:xx:xx:xx]
HOSTNAME=PCWb1
LIGHTDM_CONF="greeter-show-manual-login=true
greeter-hide-users=false
autologin-user=Wb1
autologin-user-timeout=7"

The username is now in ltsp.conf of the client, but there is still no autologin.
And there are now duplicate entries in /etc/lightdm/lightdm.conf after
sudo ltsp initrd
and restarting the client, and I don't know why.
I'm gradually resigning.

**/etc/lightdm/lightdm.conf**
\# Generated by `ltsp init`, see man:ltsp(8)
\# You can append content here by specifying the LIGHTDM_CONF parameter
[Seat:*]
\# Work around https://github.com/CanonicalLtd/lightdm/issues/49
greeter-show-manual-login=true
greeter-hide-users=false
autologin-user=
autologin-user-timeout=2
greeter-show-manual-login=true
greeter-hide-users=false
autologin-user=Wb1
autologin-user-timeout=7
